galgebra/mv.py

```diff
@@ -15,13 +15,18 @@
 )
 from sympy import exp as sympy_exp
 from sympy import N as Nsympy
+from sympy.printing.latex import LatexPrinter as _LatexPrinter
+from sympy.printing.str import StrPrinter as _StrPrinter
 
 from . import printer
 from . import metric
 from .printer import ZERO_STR
 from ._utils import KwargParser as _KwargParser
 from . import dop
 
+if TYPE_CHECKING:
+    from galgebra.ga import Ga
```

Comment:
What purpose does this line serve?

Reply:
`Ga` needs to be imported in order to use as a type annotation - but if we import it, we'd end up with a circular import. https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING is a special marker to indicate that the variable is there for `flake8` and `mypy`, but not at runtime.

Reply:
So the fact is these function actually is relying on circular import, we need to move them out someday ( I move a few back then to break circular dependency but these seem to be the ones I missed)

Reply:
No, the function isn't relying on a circular import - just the type annotations are. That's to be expected in my opinion.

The `Ga` class _creates_ `Mv` instances, so needs a direct import of the type. However, the `Mv` class only  _uses_ `Ga` instances passed in from elsewhere, so only needs `Ga` "imported" as a documentation hint.

Reply:
I see